Reproducibility/src/main/java/org/tribuo/reproducibility/ReproUtil.java

```diff
@@ -196,52 +232,41 @@ private ReproUtil(ModelProvenance provenance, Model<?> originalModel) {
 
         // If the data source can be recovered from cm, override the properties before the dataSource is
         // instantiated, and then instantiate the source.
-        DataSource<T> dataSource = null;
-        if (sourceName != null){
-            dataSource  = (DataSource<T>) cm.lookup(sourceName);
+        if (sourceName == null){
+            throw new IllegalStateException("Failed to find source with expected class " + dataSourceClass);
         }
 
-        // Throw error if the source cannot be instantiated.
-        if (dataSource == null) {
-            throw new IllegalArgumentException("The provided provenance has no data source");
+        // Type is restricted by source construction
+        @SuppressWarnings("unchecked")
+        DataSource<T> dataSource = (DataSource<T>) cm.lookup(sourceName);
```

Comment:
As above
```suggestion
        D dataSource = (D) cm.lookup(sourceName);
```

Reply:
As we have the class, I used `dataSourceClass.cast(cm.lookup(sourceName))` which allowed me to drop the warning suppression.